src/Services/ClientRegistry.php

```diff
@@ -64,6 +69,16 @@ private function buildClientConfiguration(array $conf): ClientConfiguration
             $conf['uri'] = self::buildConnectionUri($conf['hosts']);
         }
 
+        $conf['uriOptions'] = [
+            'replicaSet' => $conf['replicaSet'],
+            'ssl' => $conf['ssl'],
+            'connectTimeoutMS' => $conf['connectTimeoutMS'],
+            'readPreference' => $conf['readPreference'],
+        ];
+        if ($this->uriOptionsService instanceof UriOptionsInterface) {
+            $conf['options'] = $this->uriOptionsService->buildUriOptions($conf['uriOptions']);
```

Comment:
Should it be 
```php
$conf['uriOptions'] = $this->uriOptionsService->buildUriOptions($conf['uriOptions']);
```
instead?

I see you're using `$conf['uriOptions']` at line 92